### PR TITLE
add sso using okta docs

### PR DESCRIPTION
**Pull request by dushansilva**
This will add docs to SSO using OKTA as an external IDP with OIDC/SAML

### CODE REVIEW

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+In the following document we will explain how to connect OKTA as a third party Identity provier to WSO2 API-Manager. Before we start first make sure you have all the pre-requisite mentioned below.
+
+### Pre-requisites
+
+1. Create an account in [https://developer.okta.com/](https://developer.okta.com/)
+2. Download the latest API-Manager distribution from [https://wso2.com/api-management/](https://wso2.com/api-management/)
```
Comment (isharac) [inline en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md:6]: Please check for the similar comments in above doc

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
+In the following document we will explain how to connect OKTA as a third party Identity provier to WSO2 API-Manager. Before we start first make sure you have all the pre-requisite mentioned below.
+
+### Pre-requisites
+
+1. Create an account in [https://developer.okta.com/](https://developer.okta.com/)
+2. Download the latest API-Manager distribution from [https://wso2.com/api-management/](https://wso2.com/api-management/)
+3. Unzip the distribution and open the `deployment.toml` file located in `<APIM_HOME>/repository/conf/` and add the following configuration
```
[the remark this answers is not in the document]
Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md:7]: If t his is not enabled email username will fail. Tried it and seems like it does not work without enabling. If it is enabled both normal and email works

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
+In the following document we will explain how to connect OKTA as a third party Identity provier to WSO2 API-Manager. Before we start first make sure you have all the pre-requisite mentioned below.
+
```
[the remark this answers is not in the document]
Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md:2]: fixed https://github.com/wso2/docs-apim/pull/965/commits/35a30b0ab31168b682968dd1831c83ac5eb0d731

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+1. Go to the OKTA admin portal and navigate to Applications -> Add Application
+[![]({{base_path}}/assets/img/learn/okta-add-new-application.png)]({{base_path}}/assets/img/learn/okta-add-new-application.png)
+
+2. Select type web and use the following details
+
+    [![]({{base_path}}/assets/img/learn/okta-add-new-application-web.png)]({{base_path}}/assets/img/learn/okta-add-new-application-web.png)
+
+    [![]({{base_path}}/assets/img/learn/okta-add-new-application-details.png)]({{base_path}}/assets/img/learn/okta-add-new-application-details.png)
+3. Next we need to add a new attribute to the default user profile of OKTA represent a the user role. Navigate to Users -> Profile Editor and click on the pencil icon to edit the default profile
```
[the remark this answers is not in the document]
Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md]: fixed https://github.com/wso2/docs-apim/pull/965/commits/35a30b0ab31168b682968dd1831c83ac5eb0d731

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+    [![]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit1.png)]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit1.png) 
+    <br/>
+    <br/>
+    [![]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit2.png)]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit2.png) 
+    <br/>
+    <br/>
+    [![]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit3.png)]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit3.png) 
+
+    This will allow the user a user having the okta_role to login to publisher and devportal
```
[the remark this answers is not in the document]
Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md]: fixed https://github.com/wso2/docs-apim/pull/965/commits/35a30b0ab31168b682968dd1831c83ac5eb0d731

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+6. Expand Role configuration and add the following role. Here we check if the user that is being logged in has the role `any` and assign him the local role okta_role
+
+    <img src="{{base_path}}/assets/img/learn/okta-apim-role-oidc-role-mapping.png"/>
+
+7. Enable Just in time provisioning so that the user will be saved in the API-Manager user store
+
+    <img src="{{base_path}}/assets/img/learn/okta-apim-role-oidc-jit.png"/>
+
+8. Navigate to Service providers -> list as shown below. There are two service providers created apim_publisher, apim_devportal. Click on edit on publisher.
```
[the remark this answers is not in the document]
Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md]: fixed https://github.com/wso2/docs-apim/pull/965/commits/35a30b0ab31168b682968dd1831c83ac5eb0d731

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+    !!!warning
+        You will have to logged into the devportal and publisher at least once for the two service providers to appear as it is created during first login.
```
[the remark this answers is not in the document]
Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md]: fixed https://github.com/wso2/docs-apim/pull/965/commits/35a30b0ab31168b682968dd1831c83ac5eb0d731

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+
+    <img src="{{base_path}}/assets/img/learn/okta-apim-role-oidc-sp.png"/>
+
+    Expand local and outbound authentication configuration and under federated authentation select the name of the identity provider you created.
```
[the remark this answers is not in the document]
Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md]: fixed https://github.com/wso2/docs-apim/pull/965/commits/35a30b0ab31168b682968dd1831c83ac5eb0d731

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+
+    <img src="{{base_path}}/assets/img/learn/okta-apim-role-oidc-sp-outbound.png"/>
+    
+    Repeat the same for devportal service provider.
```
[the remark this answers is not in the document]
Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md]: fixed https://github.com/wso2/docs-apim/pull/965/commits/35a30b0ab31168b682968dd1831c83ac5eb0d731

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+
+    add the following role value. This will be used in the API-Manager to map an internal role to user that will be provisioned.
+    <img src="{{base_path}}/assets/img/learn/okta-profile-edit3.png"/>
+
+
+### Setup API-Manager
+
+1. Login in to `https://localhost:9443/carbon`.
+
+2. First we need to create a role that needs to be assinged to users that will be provioned from okta. click on add in Users and Roles section and add a new role.
```
[the remark this answers is not in the document]
Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md]: fixed https://github.com/wso2/docs-apim/pull/965/commits/35a30b0ab31168b682968dd1831c83ac5eb0d731

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (isharac) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-oidc.md:7]: +1

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    <img src="{{base_path}}/assets/img/learn/okta-apim-role-oidc-role-mapping.png"/>
+
+7. Enable Just in time provisioning so that the user will be saved in the API-Manager user store
+
+    <img src="{{base_path}}/assets/img/learn/okta-apim-role-oidc-jit.png"/>
+
+8. Navigate to Service providers -> list as shown below. There are two service providers created apim_publisher, apim_devportal. Click on edit on apim_publisher.
+
+    !!!warning
+        You will have to logged into the Developer Portal and publisher at least once for the two service providers to appear as it is created during first login.
```
Comment (isharac) [inline en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md:153]: publisher/Publisher

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+    <br/>
+    <br/>
+    [![]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit2.png)]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit2.png) 
+    <br/>
+    <br/>
+    [![]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit3.png)]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit3.png) 
+
+    This will allow the user a user having the okta_role to login to Publisher and Developer Portal
+
+4. Login in to `https://localhost:9443/carbon` & Click on add in identity providers section. Enter Identity Provider Name. Expand the federated authenticators -> SAML2 Web SSO Configuration section and under Select Mode select Metadata File Configuration and update the xml file that was saved during the okta setup process.
```
Comment (isharac) [inline en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md:112]: better if we highlight texts extracted from the UI.

Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md:112]: > highlight texts extracted from the UI.

Didn't get what did you mean here

Comment (isharac) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md:112]: ex:
Click on **Add** in  **Identity Providers** section. Enter **Identity Provider Name** etc

Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md]: fixed https://github.com/wso2/docs-apim/pull/965/commits/2fc1ae7e318fc9d5bc274b7a3d09e86158228e7c

File: en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md (hunk 11 above)
Comment (dushansilva) [reply en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md]: fixed https://github.com/wso2/docs-apim/pull/965/commits/2fc1ae7e318fc9d5bc274b7a3d09e86158228e7c

**Pull request merged**